### PR TITLE
style(sqlair): use SQLair in the flag domain

### PR DESCRIPTION
**Pull request by Aflynn50**
Replace uses of StdTxn with Txn in the flag domain.

This involved adding a new `dbFlag` type and modifying the queries

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing

## QA steps
```
TEST_PACKAGES="./domain/flag/state/... -count=1 " TEST_FILTER="Suite" make run-go-tests
```

## Links

**Jira card:** [JUJU-5628](https://warthogs.atlassian.net/browse/JUJU-5628)

[JUJU-5628]: https://warthogs.atlassian.net/browse/JUJU-5628?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: domain/flag/state/state.go [hunk 1]
```diff
@@ -30,66 +30,73 @@ func NewState(factory coredb.TxnRunnerFactory, logger logger.Logger) *State {
 
 // SetFlag sets the value of a flag.
 // Description is used to describe the flag and it's potential state.
```
Comment (gfouillet) [inline domain/flag/state/state.go:32]: **typo:** Not in your changelog but... There is actually a mispell here :) `it's` should be `its`

File: domain/flag/state/state.go [hunk 2]
```diff
@@ -30,66 +30,73 @@ func NewState(factory coredb.TxnRunnerFactory, logger logger.Logger) *State {
...
+func (s *State) SetFlag(ctx context.Context, flagName string, value bool, description string) error {
 	db, err := s.DB()
 	if err != nil {
 		return errors.Trace(err)
 	}
 
-	query := `
-INSERT INTO flag (name, value, description)
-VALUES (?, ?, ?)
+	f := dbFlag{Name: flagName, Value: value, Description: description}
```
Comment (gfouillet) [inline domain/flag/state/state.go:39]: **suggestion:** `f` may be too short as name for a variable with a scope as large as a function. Moreover, in idiomatic golang, [we tend to use one-char name for receiver](https://go.dev/wiki/CodeReviewComments#receiver-names).

A good reading about naming variables: https://google.github.io/styleguide/go/decisions#variable-names

side note: This comment can be done for other usage of `f` for flag, in a lesser extent, for `stmt` as `statement`.

File: domain/flag/state/state.go [hunk 3]
```diff
@@ -30,66 +30,73 @@ func NewState(factory coredb.TxnRunnerFactory, logger logger.Logger) *State {
...
+		var outcome sqlair.Outcome
+		err := tx.Query(ctx, stmt, f).Get(&outcome)
 		if err != nil {
 			return errors.Trace(err)
 		}
-		if affected, err := result.RowsAffected(); err != nil {
+		if affected, err := outcome.Result().RowsAffected(); err != nil {
 			return errors.Trace(err)
 		} else if affected != 1 {
 			return errors.Errorf("unexpected number of rows affected: %d", affected)
```
Comment (gfouillet) [inline domain/flag/state/state.go:60]: **suggestion(not-in-changelog):** In such error messages, having an indication of how much rows where expected to be affected can be handy.

```go
return errors.Errorf("unexpected number of rows affected: %d (should be 1)", affected)
```

**Review by gfouillet — COMMENTED** [review]
Look good for me.

**Review by SimonRichardson — APPROVED** [review]
Might be worth fixing @gfouillet changes whilst you're here.

File: domain/flag/state/state.go (hunk 2 above)
Comment (Aflynn50) [reply domain/flag/state/state.go:39]: Ah yea, had meant to change this to `flag` after chaining the parameter name from `flag` to `flagName`, thanks

File: domain/flag/state/state.go (hunk 3 above)
Comment (Aflynn50) [reply domain/flag/state/state.go:60]: Nice, will add that

File: domain/flag/state/state.go (hunk 2 above)
Comment (gfouillet) [reply domain/flag/state/state.go]: You miss an occurence l. 80 ;)

Comment (Aflynn50) [reply domain/flag/state/state.go]: Got it, thanks!

**Pull request merged**